### PR TITLE
sandbox mode: switch to use juju2 API and convert all API calls.

### PR DESCRIPTION
**Pull request by frankban**
Also remove support for GUI server Deployer related API calls,
no longer in use.

### CODE REVIEW

File: jujugui/static/gui/src/app/store/env/fakebackend.js [hunk 1]
```diff
@@ -511,7 +506,7 @@ YUI.add('juju-env-fakebackend', function(Y) {
       var response = {};
       // Add units if requested.
       if (unitCount !== 0) {
-        response = this.addUnit(options.name, unitCount, options.toMachine);
```
Comment (kadams54) [inline jujugui/static/gui/src/app/store/env/fakebackend.js]: Out of curiosity, is this part of the current API changes, or cleanup?

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5303/
Test PASSed.

File: jujugui/static/gui/src/app/store/env/fakebackend.js (hunk 1 above)
Comment (frankban) [inline jujugui/static/gui/src/app/store/env/fakebackend.js]: It's cleaning up of an argument that we don't use (and that was broken anyway).

File: jujugui/static/gui/src/test/test_sandbox_go.js [hunk 2]
```diff
@@ -20,20 +20,18 @@ with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 (function() {
 
-  describe('sandbox.GoJujuAPI', function() {
+  describe('sandbox.JujuAPI', function() {
```
Comment (kadams54) [inline jujugui/static/gui/src/test/test_sandbox_go.js]: I'm assuming this change is because there's no longer anything *but* the Go Juju API?

Comment (frankban) [inline jujugui/static/gui/src/test/test_sandbox_go.js]: Yes, indeed.

**Comment by kadams54** [comment]
:+1: Nice work! So nice to see all the code cleanup.

**Comment by frankban** [comment]
I am shipping this now to unblock other work.
Thanks for the review Kyle!
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**